src/mlpack/core/util/split_data.hpp

```diff
@@ -0,0 +1,108 @@
+#ifndef __MLPACK_CORE_UTIL_SPLIT_DATA_HPP
+#define __MLPACK_CORE_UTIL_SPLIT_DATA_HPP
+
+#include <mlpack/core.hpp>
+
+#include <iterator>
+#include <numeric>
+#include <random>
+#include <tuple>
+#include <vector>
+
```

Comment:
It looks like, since you striped the code, we can get rid of almost all of the header files except for the tuple header?